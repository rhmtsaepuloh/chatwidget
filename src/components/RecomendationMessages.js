import React from "react";
import Carousel from "nuka-carousel";
import { MdKeyboardArrowLeft, MdKeyboardArrowRight } from "react-icons/md";
import isEmpty from "lodash/isEmpty";

import "../styles/RecomendationMessages.scss";

const RecomendationMessages = ({
  recomendedData,
  onPostBack,
  payload = null,
  urlImagePreview,
}) => {
  const ButtonCarousel = ({ text, key }) => {
    return (
      <div
        className='button'
        key={key}
        onClick={() => onPostBack(text, payload, "text")}>
        <p className='text m-0 truncate'>{text}</p>
      </div>
    );
  };
  return isEmpty(recomendedData) ? null : (
    <div
      className='recomendation w-full rounded box-border p-2'
      style={{ zIndex: isEmpty(urlImagePreview) ? 1 : -1 }}>
      <Carousel
        defaultControlsConfig={{
          pagingDotsStyle: {
            display: "none",
          },
        }}
        wrapAround={true}
        slidesToShow={2.5}
        renderCenterLeftControls={({ previousSlide }) => (
          <div
            className='arrow-button-left bg-white slide-visible'
            onClick={previousSlide}>
            <MdKeyboardArrowLeft className='text-gray-400 text-4xl' />
          </div>
        )}
        renderCenterRightControls={({ nextSlide }) => (
          <div
            className='arrow-button-right bg-white slide-visible'
            onClick={nextSlide}>
            <MdKeyboardArrowRight className='text-gray-400 text-4xl' />
          </div>
        )}>
        {recomendedData.map((item, i) => (
          <ButtonCarousel key={i} text={item.message} />
        ))}
      </Carousel>
    </div>
  );
};

export default RecomendationMessages;
