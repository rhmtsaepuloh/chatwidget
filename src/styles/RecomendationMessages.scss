.recomendation {
  position: fixed;
  bottom: 70px;
  background: #fff;
  padding: 8px;
  border-bottom: solid 1px #f3f3f3;
  height: 50px;
  @media (max-width: 768px) {
    width: 100% !important;
  }
  @media (min-width: 768px) {
    width: 378px !important;
  }
  .button {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    padding: 4px 10px;
    background: #f2f6fd;
    /* Chat Widget / WHITE 60 */
    border: 1px solid #e0e0e0;
    box-sizing: border-box;
    border-radius: 16px;
    /* Inside Auto Layout */
    flex: none;
    order: 1;
    flex-grow: 0;
    margin: 0px 5px;
    .text {
      font-family: Roboto;
      font-style: normal;
      font-weight: 400;
      font-size: 14px;
      color: #4f4f4f;
    }
  }
  .slider-frame {
    padding-top: 2px !important;
    padding-left: 25px !important;
  }
  .arrow-button-left {
    height: 40px;
    margin-top: -12px;
    margin-left: -10px;
  }
  .arrow-button-right {
    height: 40px;
    margin-top: -12px;
    margin-right: -10px;
  }
  .slider {
    outline: none !important;
  }
}
