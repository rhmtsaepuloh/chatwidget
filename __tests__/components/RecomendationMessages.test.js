/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import RecomendationMessages from "../../src/components/RecomendationMessages";

const props = {
  onPostBack: jest.fn(),
  payload: null,
  urlImagePreview: "",
};

describe("render correctly recomendation messages components", () => {
  test("render if recomended data is empty", () => {
    const wrapper = shallow(
      <RecomendationMessages {...props} recomendedData={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("render if recomended data not empty", () => {
    const recomendedData = [
      {
        id: 2,
        message: "qwerty",
        clientId: 93,
        createdAt: "2021-04-01T04:54:08.482Z",
        updatedAt: "2021-04-01T04:54:18.919Z",
      },
    ];
    const wrapper = shallow(
      <RecomendationMessages {...props} recomendedData={recomendedData} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
