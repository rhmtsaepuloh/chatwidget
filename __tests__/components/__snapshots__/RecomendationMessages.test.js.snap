// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly recomendation messages components render if recomended data is empty 1`] = `ShallowWrapper {}`;

exports[`render correctly recomendation messages components render if recomended data not empty 1`] = `ShallowWrapper {}`;
